### PR TITLE
jujutsu: 0.9.0 -> 0.10.0

### PR DESCRIPTION
**Pull request by mfrw**
## Description of changes
Diff: https://github.com/martinvonz/jj/compare/v0.9.0...v0.10.0
Changelog: https://github.com/martinvonz/jj/blob/v0.10.0/CHANGELOG.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cafkafk** [comment]
Result of `nixpkgs-review pr 259151` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>jujutsu</li>
  </ul>
</details>

**Review by cafkafk — APPROVED** [review]
```
❯ jj git clone git@github.com:martinvonz/jj.git
Fetching into new repo in "/home/ces/.cache/nixpkgs-review/pr-259151/jj"
Working copy now at: ywrtwzqu 4586d1a9 (empty) (no description set)
Parent commit      : ksnuryps ce933507 main | release: release version 0.10.0
Added 267 files, modified 0 files, removed 0 files
```

File: pkgs/applications/version-management/jujutsu/default.nix [hunk 1]
```diff
@@ -20,16 +20,16 @@
...
   pname = "jujutsu";
-  version = "0.9.0";
+  version = "0.10.0";
 
   src = fetchFromGitHub {
     owner = "martinvonz";
     repo = "jj";
     rev = "v${version}";
-    sha256 = "sha256-5RN2xaH591/83iNXRcW9i/TyU5ndPZq3P/BesHM9I6w=";
+    sha256 = "sha256-LJW4Px3K5cz6RJ4sUbwUXsp2+rzEW5wowi+DALHajYA=";
```
Comment (cafkafk) [inline pkgs/applications/version-management/jujutsu/default.nix:29]: (non-blocking: you could change this to `hash`)
```suggestion
    hash = "sha256-LJW4Px3K5cz6RJ4sUbwUXsp2+rzEW5wowi+DALHajYA=";
```

Comment (mfrw) [reply pkgs/applications/version-management/jujutsu/default.nix:29]: sure will do :)

**Review by thoughtpolice — APPROVED** [review]

**Pull request merged**